src/ceph-disk/ceph_disk/main.py

```diff
@@ -3756,6 +3756,25 @@ def main_activate(args):
             )
             osd_data = get_mount_point(cluster, osd_id)
 
+            args.cluster = cluster
+
+            # Check if encrypted block device in journal
+
+            block_dmcrypt_path = os.path.join(osd_data, 'block_dmcrypt')
+
+            if args.dmcrypt and os.path.exists(block_dmcrypt_path):
+                partition = DevicePartition.factory(
+                    path=None,
+                    dev=os.path.join(osd_data, 'block_dmcrypt'),
+                    args=args)
+
+                partition.osd_dm_key = get_dmcrypt_key(
```

Comment:
also, shall we fix `mount_activate()` instead? because it is used by both `main_activate_space()` and `main_activate()`.  there are chances that our user want to activate the OSD using a partition, instead of using a whole block device.